proto: allow rustls::ClientConfig without initial suite

djc opened the pull request.

djc commented:
Hmm, maybe we should also implement `TryFrom` for `Arc<rustls::ClientConfig>`?

Ralith commented:
> Hmm, maybe we should also implement TryFrom for Arc<rustls::ClientConfig>?

Oops, yes, I think we should have done this from the start. Luckily I don't *think* there's a drawback to having both... maybe slightly worse inference in rare cases.

djc commented:
> Oops, yes, I think we should have done this from the start. Luckily I don't _think_ there's a drawback to having both... maybe slightly worse inference in rare cases.

Added this in a second commit.

Also changed the first commit to actually check that the specified `Suite` is actually AES-128-GCM-SHA256.
